lib/Serialization/ModuleFile.cpp

```diff
@@ -959,6 +959,36 @@ Optional<CommentInfo> ModuleFile::getCommentForDecl(const Decl *D) const {
   return getCommentForDeclByUSR(USRBuffer.str());
 }
 
+void ModuleFile::collectBasicSourceFileInfo(
+    llvm::function_ref<void(const BasicSourceFileInfo &)> callback) const {
+  if (Core->SourceFileListData.empty())
+    return;
+  assert(!Core->SourceLocsTextData.empty());
+
+  auto *Cursor = Core->SourceFileListData.bytes_begin();
+  auto *End = Core->SourceFileListData.bytes_end();
+  while (Cursor < End) {
+    auto fileID = endian::readNext<uint32_t, little, unaligned>(Cursor);
+    auto fpStr = StringRef{reinterpret_cast<const char *>(Cursor),
```

Comment:
Heh, this took me a second to understand. Maybe worth a comment that this is a fixed-length string?